third_party/terraform/resources/resource_dataflow_job.go

```diff
@@ -499,3 +507,36 @@ func resourceDataflowJobIsVirtualUpdate(d *schema.ResourceData) bool {
 
 	return false
 }
+
+func waitForDataflowJobToBeUpdated(d *schema.ResourceData, config *Config, replacementJobID string, timeout time.Duration) error {
+	return resource.Retry(timeout, func() *resource.RetryError {
+		project, err := getProject(d, config)
+		if err != nil {
+			return resource.NonRetryableError(err)
+		}
+
+		region, err := getRegion(d, config)
+		if err != nil {
+			return resource.NonRetryableError(err)
+		}
+
+		replacementJob, err := resourceDataflowJobGetJob(config, project, region, replacementJobID)
+		if err != nil {
+			if isGoogleApiErrorWithCode(err, 404) {
+				return resource.RetryableError(fmt.Errorf("the replacement job with ID %q was not found.", replacementJobID))
+			}
+			return resource.NonRetryableError(err)
+		}
+
+		state := replacementJob.CurrentState
+		switch state {
+		case "JOB_STATE_PENDING":
+			return resource.RetryableError(fmt.Errorf("the replacement job with ID %q has not yet started and has state %q.", replacementJobID, state))
+		case "JOB_STATE_FAILED":
+			return resource.NonRetryableError(fmt.Errorf("the replacement job with ID %q failed with state %q.", replacementJobID, state))
+		default:
```

Comment:
```suggestion
		case "":
			return resource.RetryableError(fmt.Errorf("the replacement job with ID %q does not have a defined state. Retrying.", replacementJobID, state))
		default:
```
Found a case where the state just returns empty before eventually getting to `JOB_STATE_FAILED`. Adding a retry here gets to the failed state

Reply:
we can also combine this with "JOB_STATE_PENDING" and change the message to "has pending state %q"

Reply:
Great catch! Fixed.